flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java

```diff
@@ -619,7 +624,25 @@ public void alterPartitionColumnStatistics(ObjectPath tablePath, CatalogPartitio
   @Override
   public List<CatalogPartitionSpec> listPartitions(ObjectPath tablePath)
       throws CatalogException {
-    throw new UnsupportedOperationException();
+    Table table = icebergCatalog.loadTable(toIdentifier(tablePath));
```

Comment:
Better to use the `FlinkCatalog#loadIcebergTable`,  because it's wrapped to throw a `TableNotExistException` for flink.

Reply:
I updated it